orc: Add support for @1.6.13 to get getSoftwareVersion

payerle opened the pull request:
See #43221 --- arrow +orc wants getSoftwareVersion which is only added in orc@1.6.11

The thirdparty.patch has been updated to something that will work with both 1.6.5 and 1.6.11.

Unfortunately, upgrading to @1.7: will need more work

<!--  
Remember that `spackbot` can help with your PR in multiple ways:
- `@spackbot help` shows all the commands that are currently available
- `@spackbot fix style` tries to push a commit to fix style issues in this PR
- `@spackbot re-run pipeline` runs the pipelines again, if you have write access to the repository 
-->

payerle committed:
orc: Add support for @1.6.13 to get getSoftwareVersion

See #43221 --- arrow +orc wants getSoftwareVersion which is only added in orc@1.6.11

The thirdparty.patch has been updated to something that will work with both 1.6.5
and 1.6.11.

Unfortunately, upgrading to @1.7: will need more work

haampie commented on var/spack/repos/builtin/packages/orc/package.py:
```diff
@@ -27,7 +28,7 @@ class Orc(CMakePackage):
     depends_on("lz4@1.7.5:")
 
-    patch("thirdparty.patch")
+    patch("thirdparty.patch", when="@1.6")
```
Can you change the `thirdparty.patch` filename? Otherwise patch cache issues can happen.

bernhardkaindl replied:
> Can you change the thirdparty.patch filename? Otherwise patch cache issues can happen.

@payerle: Can you apply the change that haampie requested?

haampie commented:
Please add a `depends_on("orc@1.6.13:", when="+orc") lowerbound in `arrow`

payerle committed:
orc: changed name of patchfile

changed name of patch file as per suggestion by @haampie

bernhardkaindl reviewed (approved):
The review comment is applied.

bernhardkaindl reviewed (changes requested):
@payerle: 1 of 5 hunks in the new patch do not apply to 1.6.13, fix it.

payerle committed:
orc: Fix thirdparty-1.6.patch

Somehow a trailing whitespace got into the file, causing patching to fail.
Fixed.

payerle committed:
arrow: Merging changes from PR #43223 into #43222

arrow+orc needs getSoftwareVersion from orc, which was added in orc@1.6.11
This places a minimum on orc version required for arrow +orc

Merging what was PR #43223 into this PR as the new orc version was
added just for this.

payerle committed:
arrow: Fix version in orc dependency

I previously misread error message thinking it was complaining that
no orc@1.6.11 was defined, but only complaining that no versions of
orc satisfying orc@1.6.11: were found.
So can leave the more accurate orc@1.6.11: requirement as this
PR adds 1.6.13
